### ISSUE TITLE
Expose all metatables

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#11639]
## Problem

Meta- or methodtables like that of ItemStack or ObjectRefs have to be extracted in odd ways (usually by obtaining an instance and then getting the metatable). This requires ugly hacks: While obtaining an ItemStack is trivial, obtaining an ObjectRef requires some hacks (like waiting for a player to join).

## Solution

Just expose meta- and methodtables as global variables.

**Comment by Desour** [comment]
Are you sure you mean the metatable? What you get via `getmetatable` is the `__index` field of the metatable:
https://github.com/minetest/minetest/blob/fd8a8501bc26dfca2a93d51000867b8592210040/src/script/lua_api/l_object.cpp#L2286-L2288

**Comment by appgurueu** [comment]
> Are you sure you mean the metatable? What you get via `getmetatable` is the `__index` field of the metatable:

Yeah, that's what I refer to as "methodtable". I'm mostly fine with that too.

> what would you need the metatable for?

Common use case is hooking ObjectRef functions. I was thinking of caching object properties in a weak table for instance. Or updating an entities HP bar if `set_hp` is called.

**Comment by AFCMS** [comment]
Would be very useful for MineClone2 :+1:
